### DISCUSSION TITLE
Does Jacobian important for the computation?

### DISCUSSION THREAD

**Discussion by Oops-Qiao** [idaholab/moose#21958] [Q&A General] [answered]
Hello buddy,

I want to ask that in the kernel, if we donot use the AD version, then we should write the weak form and Jacobian. Today I heard from my friend that the weak form should be well written, however, the Jacobian can be written with some terms igonored. Can anyone make comment for this? how and why.

Thank you in advance.

Kind regards,
Qiao

**Comment by GiudGiud** [comment, marked as answer]
Hello

Is this in the context of using Newton's method or PJFNK?
With PJFNK, the jacobian may be used for preconditioning. Imperfect preconditioning may or may not affect convergence. If the jacobian is mostly correct, especially if there are correct diagonally dominant terms, then it's likely to converge anyway.

> Hello buddy,

who is this buddy you refer to?

Guillaume

**Reply by Oops-Qiao** [reply]
Thank you very much. I mean in the context of Newton method.

**Reply by Oops-Qiao** [reply]
So what you mean here is in PJFNK, incorrect Jacobian just affect the convergence, not the solution right?

**Reply by Oops-Qiao** [reply]
Hello buddy means Hello friend, am I right?

**Reply by GiudGiud** [reply]
in both methods, an imperfect Jacobian will only affect the convergence, not the solution (if it is unique). You should search the litterature to learn more about convergence with imperfect Jacobians, we dont have much in the documentation. 

The solution, which in a well-posed problem is unique, is uniquely defined such that the residual is 0.

Buddy is pretty familiar for addressing other scientists who you do not know personally ;).

**Reply by Oops-Qiao** [reply]
Ok, thank you very much and also for the explaination of buddy. ：）
